synapse/federation/transport/server/_base.py

```diff
@@ -279,29 +277,19 @@ async def new_func(
                 logger.warning("authenticate_request failed: %s", e)
                 raise
 
-            request_tags = {
-                SynapseTags.REQUEST_ID: request.get_request_id(),
-                tags.SPAN_KIND: tags.SPAN_KIND_RPC_SERVER,
-                tags.HTTP_METHOD: request.get_method(),
-                tags.HTTP_URL: request.get_redacted_uri(),
-                tags.PEER_HOST_IPV6: request.getClientIP(),
-                "authenticated_entity": origin,
-                "servlet_name": request.request_metrics.name,
-            }
-
-            # Only accept the span context if the origin is authenticated
-            # and whitelisted
+            # update the active opentracing span with the authenticated entity
+            set_tag("authenticated_entity", origin)
+
+            # if the origin is authenticated and whitelisted, link to its span context
             context = None
             if origin and whitelisted_homeserver(origin):
                 context = span_context_from_request(request)
 
-            scope = start_active_span(
-                "incoming-federation-request", child_of=context, tags=request_tags
+            scope = start_active_span_follows_from(
+                "incoming-federation-request", contexts=(context,) if context else ()
             )
```

Comment:
this changes the "incoming-federation-request" span so that its parent is no longer the outgoing request of the remote server, but rather the regular per-incoming-request span that is started in `AsyncResource`. Instead, the outgoing request span is set as a "reference" on the incoming request span.

Given that it's extremely unusual for us to have a single Jaeger instance which contains two federating servers, this seems by far the more useful approach.